test/kldivergences.jl

```diff
@@ -0,0 +1,76 @@
+@testset "KL divergences" begin
+    function logdiff(P, Q, x)
+        return let p = pdf(P, x)
+            (p > 0) * log(p / pdf(Q, x))
+        end
```

Comment:
The `let` block is not needed/useful here, I think. You could also use the `logpdf` version to handle small probabilities better.
```suggestion
        p = pdf(P, x)
        return (p > 0) * log(p / pdf(Q, x))
```
or
```suggestion
        logp = logpdf(P, x)
        return (logp > -Inf) * (logp - logpdf(Q, x))
```